store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/RealStore.kt

```diff
@@ -85,10 +85,24 @@ internal class RealStore<Key : Any, Network : Any, Output : Any, Local : Any>(
                 }
             }
 
-            cachedToEmit?.let { it: Output ->
+            if (cachedToEmit != null) {
                 // if we read a value from cache, dispatch it first
-                emit(StoreReadResponse.Data(value = it, origin = StoreReadResponseOrigin.Cache))
+                emit(
+                    StoreReadResponse.Data(
+                        value = cachedToEmit,
+                        origin = StoreReadResponseOrigin.Cache
+                    )
+                )
+                if (!request.fetch) {
+                    // This request was only for cached items, so we stop here
+                    return@flow
+                }
+            } else if (!request.fetch) {
+                // The cache is empty or invalid but we don't want to hit the fetcher
+                emit(StoreReadResponse.NoNewData(origin = StoreReadResponseOrigin.Cache))
+                return@flow
```

Comment:
Is the feature request for literally cache only? I understood it to be memory or SOT but not network

Reply:
Yes, I understood the feature request literally. I can add SoT checking though. Do you think it makes sense to have a cache-only option like this still or should it be just a non-network option?

Reply:
Imo cacheOnly should attempt memory cache and SoT. Because the use case is to be able to use cache if we have it, otherwise be happy with no value. One particular case we had was analytics, we would send additional data if we had it, but don't attempt to fetch because it would make no sense.

Additionally don't really find any benefit in exposing a literal (memory) cache only because Store abstracts it from consumers.